src/app/public/modules/restricted-view/restricted-view-auth.service.ts

```diff
@@ -39,11 +59,27 @@ export class SkyRestrictedViewAuthService  {
           if (typeof permissions === 'number') {
             permissions = [permissions];
           }
+
           if (permissions.indexOf(1) > -1) {
+            this.setHasBeenAuthenticated();
             this.isAuthenticated.next(true);
+            return;
           }
         }
+        // NOTE: Emitting here allows for the watching of when any response is returned.
+        this.isAuthenticated.next(false);
       },
-      () => {});
+        () => {
+          // NOTE: Emitting here allows for the watching of when any response is returned.
+          this.isAuthenticated.next(false);
```

Comment:
This might mess with the unit tests, but you could avoid some complexity here (and with all the `skip` calls) if the `isAuthenticated` property was a `Subject` instead of a `BehaviorSubject`. Did you try experimenting with that already?

Reply:
I didn't write `isAuthenticated`. It was existing - so I didn't mess with looking at changing it as it is exported by the public API and is a public property so that would technically be a breaking change.

Reply:
Can you explain why we need both of these `this.isAuthenticated.next(false);`? The comment above each one isn't clear to me.

Reply:
Can you explain why we need both of these `this.isAuthenticated.next(false);` statements? The comment above each one isn't clear to me.

Reply:
Ya. So I had to add these because the `restricted-view-auth.service` has an initial default value for the `BehaviorSubject` of false. This causes the alert to flash due to the subscription in alert component registering this false value before the actual auth check has finished. To fix this I need to skip the first value from the `isAuthenticated` `BehaviorSubject`. In doing this, I realized that we were not setting the value of `false` when we got the authentication value back (presumably because we already defaulted to false and today we only ever call that authentication check once on initialization). 

I in theory could have gotten around this in a couple ways but felt like there were cons to both of the other solutions

1. I could have changed so that the subject does not initialize to false
   - I felt that this would be a breaking change because that `BehaviorSubject` is public
   - That still would have required these new calls to set the `false` and just would have avoided the alert's `skip` operator
2. I could have delayed the subscription inside the alert
   - I felt that this would potentially still leave us open to a race condition because we would have to be guessing on the delay length and a slow network connection could still have not returned the authentication result prior to the delay being done.

Reply:
I guess I'm more wondering why we need to fire `false` at all. If the default behavior is to hide the alerts, and the only way to show the alert is to get a `true` from the auth response, in what scenario would we need to respond with `false`?

Reply:
The alert shows when auth is false - not when it is true.

Reply:
The alert shows when auth is `false` but there is a cookie saying that they have previously logged in as a BB employee. So because the `BehaviorSubject` defaults to `false` we have to be careful not to flash the alert in the time between initialization and the auth check coming back. That is why we `skip(1)` to ignore initialization and then watch for the return of `false` from the actual auth check.

Reply:
Thanks for the extra explanation, that helps. I feel like there's a better way we could do this if we started from scratch, but I'd rather not spend too much time on it.